unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/PublicChannelEntry.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class PublicChannelEntry : BaseComponentView
+{
+    [SerializeField] private Button openChatButton;
+    [SerializeField] private TMP_Text nameLabel;
+    [SerializeField] private PublicChannelEntryModel model;
+    [SerializeField] private UnreadNotificationBadge unreadNotifications;
+    private IChatController chatController;
+    private ILastReadMessagesService lastReadMessagesService;
+
+    public PublicChannelEntryModel Model => model;
+
+    public event Action OnOpenChat;
+
+    public override void Awake()
+    {
+        base.Awake();
+        openChatButton.onClick.AddListener(() => OnOpenChat?.Invoke());
+    }
+    
+    public void Initialize(IChatController chatController,
+        ILastReadMessagesService lastReadMessagesService)
+    {
+        this.chatController = chatController;
+        this.lastReadMessagesService = lastReadMessagesService;
+    }
+
+    public void Set(PublicChannelEntryModel model)
+    {
+        this.model = model;
+        RefreshControl();
+    }
```

Comment:
You can implement the interface IComponentModelConfig and replace this function by the Configure(BaseComponentModel newModel) one provided.

Reply:
i think `IComponentModelConfig` should be generic `IComponentModelConfig<T>` since we are violating a LiskovSubstitution principle here

Reply:
It would be ideal, Nico! I like the idea. Anyway, as I think this change would make us to modify a lot of other parts of code not related with your current implementation in this branch, in order to not increase the amount of changes here, I'm going to create a card task for that in order to apply this refactor in the future 👌

Reply:
Done! Here is the task to be planned in future sprints: https://app.zenhub.com/workspaces/explorer-6047bad476c3c0001942ee91/issues/decentraland/unity-renderer/2291